server/recceiver/cfstore.py

```diff
@@ -86,16 +89,22 @@ def __commit__(self, TR):
         for rid, (rname, rtype) in TR.addrec.iteritems():
             pvInfo[rid] = {"pvName":rname}            
         for rid, (recinfos) in TR.recinfos.iteritems():
-            if "properties" in  recinfos:
-                if rid in pvInfo:
-                    recProperties = recinfos["properties"]
-                    properties = []
-                    for prop in recProperties.split(","):
-                        p = prop.strip().split("=")
-                        properties.append({u'name': p[0], u'owner': owner, u'value': p[1]})
-                    pvInfo[rid]["infoProperties"] = properties
-                else:
-                    _log.error("could not find the associated record for properties")
+            # find intersection of these sets
+            recinfo_wl = [p for p in self.whitelist if p in recinfos.keys()]
+            if recinfo_wl:
+                pvInfo[rid]['infoProperties'] = list()
+                for infotag in recinfo_wl:
+                    _log.debug('INFOTAG = {}'.format(infotag))
+                    property = {u'name': infotag, u'owner': owner,
+                                u'value': recinfos[infotag]}
+                    # XXX this could raise
+                    prop_exists = self.client.find(name=pvInfo[rid]['pvName'],
+                                                   property=[(infotag, '*')])
+                    # if it doesn't exist, create placeholder for new property
+                    if not prop_exists:
+                        self.client.set(property={u'name': infotag,
+                                        u'owner': owner})
```

Comment:
Seems to me that this service should own the whitelisted properties as it will manage them.  Then it could create them on startup, avoiding the extra round trips.